conan_provider.cmake

```diff
@@ -388,12 +388,22 @@ function(conan_install)
     # Invoke "conan install" with the provided arguments
     set(CONAN_ARGS ${CONAN_ARGS} -of=${CONAN_OUTPUT_FOLDER})
     message(STATUS "CMake-Conan: conan install ${CMAKE_SOURCE_DIR} ${CONAN_ARGS} ${ARGN}")
+
+
+    # We inject to the PATH the folder where the CMake that invoked the provider lives
+    # so that in case it's not on the system path the conan install does not fail
+    set(_OLD_PATH $ENV{PATH})
+    set(ENV{PATH} "${CMAKE_DIR}:$ENV{PATH}")
```

Comment:
Re: my other comment

We could be on a linux system where cmake was installed in `/usr/bin/cmake` 
The default PATH on a typical Linux is : `/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin`

A regular user may also have things in `$HOME/.local/bin` and similar, or an active `venv` in python.

This line here would place `/usr/bin` at the beginning on the list for conan and below (anything conan invokes) - which could be a problem IF there are things that are installed in `/usr/bin` that are also in something that has higher priority (`/usr/local` or `~/.local/bin` or similar). 

I would avoid altering this priority unless strictly necessary, and only modify `PATH` if the directory that contains the CMake executable is not already in it

Reply:
> I would avoid altering this priority unless strictly necessary, and only modify PATH if the directory that contains the CMake executable is not already in it

Yes, this sounds reasonable concern. Modern CMake has ``cmake_path()`` helpers, maybe it is not difficult to check this.